File: v1/Demo/demo_get_velocity_pid.py
import sys, os
sys.path.append(os.path.join(os.path.dirname(__file__), "../"))
from Hand_Sdk.HandSocket import *
from Hand_Sdk.Handapi import *

def main():
    ip_left = "192.168.137.39" # lefthand
    # ip_right = "192.168.137.19" # righthand
    api = API(ip_left)
    velocity_pid = api.get_velocity_pid()

    if len(velocity_pid) != 18:
        logger.print_trace_error("velocity_pid failed")
    else:
        logger.print_trace("velocity_pid success", velocity_pid)

if __name__ == "__main__":
    main()

File: v1/Demo/source/demo_get_velocity_pid.cpp
#include "../include/Handapi.h"

using namespace HandProtocol;

int main()
{
    Handapi api;
    api.init(LEFT_IP);
    api.calibration();
    usleep(1000000);

    vector<float> velocity_pid = api.get_velocity_pid();
    if (velocity_pid.size() != 18)
    {
        Logger::get_instance()->print_trace_error("get velocity_pid failed\n");
        return ResultCode::FAIL;
    }
    
    std::cout<<"get velocity_pid success : ";
    for (int i = 0; i < velocity_pid.size(); ++i) {
        std::cout << velocity_pid[i] << " ";
    }
    std::cout << std::endl;


    return 0;
}